module/validation/receipt_validator.go

```diff
@@ -155,7 +155,12 @@ func (v *receiptValidator) resultChainCheck(result *flow.ExecutionResult, prevRe
 		return fmt.Errorf("missing initial state commitment in execution result %v", result.ID())
 	}
 	if !bytes.Equal(initialState, finalState) {
-		return engine.NewInvalidInputError("execution results do not form chain")
+		return engine.NewInvalidInputError(
+			fmt.Sprintf("execution results do not form chain: expecting init state %x, but got %x",
+				finalState,
+				initialState,
+			),
+		)
```

Comment:
I think you could use (?)
```suggestion
		return engine.NewInvalidInputErrorf("execution results do not form chain: expecting init state %x, but got %x",
				finalState,
				initialState,
		)
```